checker/cost.go

```diff
@@ -599,3 +605,30 @@ func (c *coster) newAstNode(e *exprpb.Expr) *astNode {
 	}
 	return &astNode{path: path, t: c.getType(e), expr: e, derivedSize: derivedSize}
 }
+
+func isScalar(t *exprpb.Type) bool {
```

Comment:
+godoc since many programmers consider strings scalars.

Reply:
Also, when discussing with @TristonianJones it looks like `protobuf.Any` and `protobuf.Value` are not supportable since it's not possible to know if the value is a scalar till runtime. We should include a note about that in the godoc..

Reply:
Gotcha, I've added documentation with a note about both `Any` and `Value`.